### DISCUSSION TITLE
Metamask issue with events: chainChanged and connect.

### DISCUSSION THREAD

**Discussion by nhatnguyen95** [WalletConnect/walletconnect-monorepo#2773] [Developer Support] [answered]
I am working on React Native and using @walletconnect/modal-react-native": "1.0.0-rc.2, when I try to connect to MetaMask and the chainId different with the chainId of activating network. `chainChanged` event didn't fire after returning to my app on **Android** (it works on iOS) so I can't know user has chosen the right network or not on **Android**.  Other event: "connect" also return an undefined chainId in ProviderInfo object after connect success and returning to my app.

Here is my code:

```
const {isConnected, provider} = useWalletConnectModal();
useEffect(() => {
    const registerEvents = () => {
      provider?.on('chainChanged', (chainId: string) => {
        setChainIdConnected(parseInt(chainId));
      });
      provider?.on('connect', ({chainId}) => {
        setChainIdConnected(parseInt(chainId));
      });
    };
    registerEvents();
  }, [provider]);
```

**Comment by crypblizz8** [comment]
@ignaciosantise can you help here?

**Reply by georgecpp** [reply]
so are you saying metamask is showing up in the list in modal?

**Reply by nhatnguyen95** [reply]
> so are you saying metamask is showing up in the list in modal?

It showing and connect successfully but some events did not emit on android if chainId to connect different with chainId of selecting network in Metamask.

**Comment by ignaciosantise** [comment, marked as answer]
Hey @nhatnguyen95 I've just reproduced the issue on Android, and apparently the problem is that MetaMask redirect's to the dapp before sending the chainChanged event. If you manually go back to MetaMask, you'll see the `chainChanged` event executing. 

And about the connect event, `({ chainId })` is undefined because the callback is returning the session object. I'll check with the team about this.

**Reply by ignaciosantise** [reply]
@nhatnguyen95 to get the active chain, you can call `provider.request({ method: 'eth_chainId' });`

**Reply by nhatnguyen95** [reply]
@ignaciosantise I 've tried with `provider.request({ method: 'eth_chainId' });` but it is still return chainId of chainId call in `connect` function. 
I also create an issue on Metamask's repo here: https://github.com/MetaMask/metamask-mobile/issues/6706
